### ISSUE TITLE
Batchnorm for GCN 

### ISSUE THREAD

**Issue by JanoschMenke** [rusty1s/pytorch_geometric#1066]
## ❓ Questions & Help

So I am not sure how I would implement a batchnorm layer if I am using a GCN. After a Convolution I would get a matrix of size `[nodes_per_graph*batchsize, features]`. But the `nodes_per_graph `differ between graphs so some batches haves more rows than others.  

Now would I still perform a normilaization on each column, even if the size differs from batch to batch?
Am I missing something obvious?

**Comment by rusty1s** [comment]
The common technique for this is just to apply `BatchNorm1d` to the node feature matrix. This way, you compute statistics for each feature across all nodes in the batch.

**Comment by JanoschMenke** [comment]
So different amount of nodes in batches is not a problem.
Intuitively I understand that normalization does not depend on the size, but most of the time batchnorm is applied the data always has the same dimensions, but that would not be an issue here?

**Comment by rusty1s** [comment]
I don't see a reason why this should be a problem :)

**Comment by JanoschMenke** [comment]
Thats true, sometimes it is actually easier than one would think :) 

Thanks

**Comment by Sara-Za** [comment]
I'm using BatchNorm1d(#of nodes in the graph) after the Convolution layers (each batch is the same graph with different features). It improves the accuracy of the training nodes but deteriorates the accuracy of the test nodes. Any idea why that's happening?

**Comment by JanoschMenke** [comment]
It could be that your train and testset distribution are too different.

**Comment by rusty1s** [comment]
You can try using batch norm with `track_running_stats=False`.

**Comment by Sara-Za** [comment]
Thanks for the responses. I have already tried track_running_stats=False and it doesn't help:(
Also, I don't think too different distributions is the case here as I have tried multiple combinations of nodes for training and testing, but the results always change in the same way.
In general, when we're using batch normalization here, it normalizes features over all the nodes of the graph (including both test and train nodes). Then, the test_mask nodes' features are already considered in normalization. Is that correct?
Also, is there any example of using Batchnorm with GCN here?
